runtime/permissions.rs

```diff
@@ -37,14 +41,16 @@ pub enum PermissionState {
 impl PermissionState {
   #[inline(always)]
   fn log_perm_access(name: &str, info: Option<&str>) {
-    debug!(
-      "{}",
-      colors::bold(&format!(
-        "{}️  Granted {}",
-        PERMISSION_EMOJI,
-        Self::fmt_access(name, info)
-      ))
-    );
+    if *DEBUG_LOG_ENABLED {
+      log::debug!(
+        "{}",
+        colors::bold(&format!(
+          "{}️  Granted {}",
+          PERMISSION_EMOJI,
+          Self::fmt_access(name, info)
+        ))
```

Comment:
> Avoids "expensive" format calls if no debug logging is enabled

I'm not sure this is the case. The `log::debug!` macro should put the expression inside an if statement when compiled https://github.com/rust-lang/log/blob/bd28b80a6d2b90cb9131e21bc602df68f4f72fdb/src/macros.rs#L34

Do you think we could make a change downstream to make this faster instead of doing this code? I feel like this should be handled by the macro instead of us doing this ourselves.

Reply:
In theory we could also set the `max_level` correctly when it's disabled, didn't inspect all the code paths to see why that's not happening.

Will pick this up again in a few days

Reply:
Will merge this and then explore a more general upstream fix

Reply:
Would you be able to add a comment saying why this is temporarily done?

Reply:
Added a TODO + rationale comment